### PR TITLE
Add supp tbl

### PR DESCRIPTION
**Pull request by statasaurus**

### CODE REVIEW

File: R/metacore.R [hunk 1]
```diff
@@ -58,6 +59,18 @@ MetaCore_initialize <- function(ds_spec, ds_vars, var_spec, value_spec, derivati
...
                  codes = "List of Codes")
 
+   private$.codelist <- codelist %>%
+      add_labs(code_id = "ID of the Code List",
+               names = "Name of the Code List",
+               type = "Code List/Permitted Values/External Library",
+               codes = "List of Codes")
+
+   private$.supp <- supp %>%
+      add_labs(dataset = "Dataset Name",
```
Comment (mstackhouse) [inline R/metacore.R:69]: @statasaurus is the assumption that the variable would also live in `var_spec` and flagged as a supp? And QORIG based on derivations?

Comment (statasaurus) [reply R/metacore.R:69]: No the QORIG would be base on the origin. You shouldn't need to have a supp flag in var_spec or value_spec cause the information will be pulled from ds_vars. The plan would be you would treat your supp variable like any other. Add the label to the label caloumn in var_spec, add information about how it is derived in derivations and any origin, codelist etc. information in the value_spec

Comment (mstackhouse) [reply R/metacore.R:69]: Before we polish this off, could you put together an example to merge the necessary info from valtools and push into the supp building functions? Before we call this final I just want to make sure that we have that metcaore -> metatools -> supp dataset workflow locked down.

Comment (statasaurus) [reply R/metacore.R:69]: ... valtools? Did you mean metatools? The example is [here](https://github.com/pharmaverse/metatools/blob/c677ee5ac3ae9abf326ee300f921cfa17727e6ab/R/supp.R#L74-L79). I have created a mock metacore sdtm object which is already in the dev branch that has some supp stuff it in. I am planning on writing a where to add my supp information vingette.

**Pull request merged**